tests/test_translate_ciphers_format.sh

```diff
@@ -0,0 +1,136 @@
+#
+# On fail, the translated cipher suite names do not match the correct ones.
+# In this case the difference will be printed in stdout.
+#
+# This files main purpose is to ensure translate_ciphers.py can take strings
+# in the expected format and return them in the format compat.sh will expect.
+
+# Ciphers that will use translate_ciphers.py
+M_CIPHERS=""
+O_CIPHERS=""
+G_CIPHERS=""
+
+# Ciphers taken directly from compat.sh
+Mt_CIPHERS=""
+Ot_CIPHERS=""
+Gt_CIPHERS=""
+
+# Initial list to be split into 3
+CIPHERS="TLS-ECDHE-ECDSA-WITH-NULL-SHA      \
+    TLS-ECDHE-ECDSA-WITH-3DES-EDE-CBC-SHA   \
+    TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA    \
+    TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA    \
+    "
+
+M_CIPHERS="$M_CIPHERS $CIPHERS"
+
+G=`python3 translate_ciphers.py g "$CIPHERS"`
+G_CIPHERS="$G_CIPHERS $G"
+
+O=`python3 translate_ciphers.py o "$CIPHERS"`
+O_CIPHERS="$O_CIPHERS $O"
+
+Mt_CIPHERS="$Mt_CIPHERS                       \
+    TLS-ECDHE-ECDSA-WITH-NULL-SHA           \
+    TLS-ECDHE-ECDSA-WITH-3DES-EDE-CBC-SHA   \
+    TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA    \
+    TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA    \
+    "
+Gt_CIPHERS="$Gt_CIPHERS                       \
+    +ECDHE-ECDSA:+NULL:+SHA1                \
+    +ECDHE-ECDSA:+3DES-CBC:+SHA1            \
+    +ECDHE-ECDSA:+AES-128-CBC:+SHA1         \
+    +ECDHE-ECDSA:+AES-256-CBC:+SHA1         \
+    "
+Ot_CIPHERS="$Ot_CIPHERS               \
+    ECDHE-ECDSA-NULL-SHA            \
+    ECDHE-ECDSA-DES-CBC3-SHA        \
+    ECDHE-ECDSA-AES128-SHA          \
+    ECDHE-ECDSA-AES256-SHA          \
+    "
+
+
+# Initial list to be split into 3
+CIPHERS="TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA256         \
+          TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA384         \
+          TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256         \
+          TLS-ECDHE-ECDSA-WITH-AES-256-GCM-SHA384         \
+         "
+
+M_CIPHERS="$M_CIPHERS $CIPHERS"
+
+G=`python3 translate_ciphers.py g "$CIPHERS"`
+G_CIPHERS="$G_CIPHERS $G"
+
+O=`python3 translate_ciphers.py o "$CIPHERS"`
+O_CIPHERS="$O_CIPHERS $O"
+
+Mt_CIPHERS="$Mt_CIPHERS                               \
+    TLS-ECDHE-ECDSA-WITH-AES-128-CBC-SHA256         \
+    TLS-ECDHE-ECDSA-WITH-AES-256-CBC-SHA384         \
+    TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256         \
+    TLS-ECDHE-ECDSA-WITH-AES-256-GCM-SHA384         \
+    "
+Gt_CIPHERS="$Gt_CIPHERS                               \
+    +ECDHE-ECDSA:+AES-128-CBC:+SHA256               \
+    +ECDHE-ECDSA:+AES-256-CBC:+SHA384               \
+    +ECDHE-ECDSA:+AES-128-GCM:+AEAD                 \
+    +ECDHE-ECDSA:+AES-256-GCM:+AEAD                 \
+    "
+Ot_CIPHERS="$Ot_CIPHERS               \
+    ECDHE-ECDSA-AES128-SHA256       \
+    ECDHE-ECDSA-AES256-SHA384       \
+    ECDHE-ECDSA-AES128-GCM-SHA256   \
+    ECDHE-ECDSA-AES256-GCM-SHA384   \
+    "
+
+# Normalise spacing
+M_CIPHERS=$( echo "$M_CIPHERS" | sed -e 's/[[:space:]][[:space:]]*/ /g' -e 's/^ //' -e 's/ $//')
```

Comment:
As the arguments being passed around are separated by an unknown number of spaces, it is necessary to remove the spacing to compare them. The code is copied from compat.sh. Also as MBedTLS ciphers are not translated, unlike OpenSSL and GnuTLS, it will maintain this inconsistent spacing.

If I am to translate this test file to python unit tests then it doesn't matter so much as I can split the strings into a list and compare them, but format.sh was originally supposed to be a test to make sure compat.sh wont have any issues, and will receive what it expects

Reply:
My point is that expanding a variable without double quotes already does this space normalization.

The similar code in `compat.sh` could be simplified in the same way.

Reply:
Ah I see, I didn't notice the removed quotes. Should I make the same change to compat.sh?

Reply:
Changing `compat.sh` is out of scope since you aren't touching that part of its code. But you can do it if you want.